### PR TITLE
fix: nameet final validations on fleet and onboard

### PR DESCRIPTION
**Pull request by nameetjambla**
In  this module all the validations and placeholders for carrier onboard and fleet module is updated

### CODE REVIEW

File: src/app/pages/fleet/assets/add-assets/add-assets.component.html [hunk 1]
```diff
@@ -496,14 +496,14 @@ <h4 class="text-4 font-weight-bold mt-0 mb-0 text-dark">{{pageTitle}}</h4>
...
-                      placeholder="eg. 345678 " name="assetDetails.licencePlateNumber"
+                      placeholder="eg. CAL 745" name="assetDetails.licencePlateNumber"
                       class="form-control" type="text" maxlength="8" #pName="ngModel" required pattern="^[a-zA-Z0-9\s]{7,8}$">
                       <div *ngIf="pName.invalid && (pName.dirty || pName.touched)"  class="text-danger">
                         <div *ngIf="pName.errors.required">
                           Licence plate number is required.
                         </div>
                           <div *ngIf="pName.errors.pattern">
-                          Licence plate number contains 7-8 uppercase, lowercase, numbers and special characters for period(space)only .
+                            Licence plate number must be between 7-8 alphanumeric characters only.
```
Comment (FH-sandeepkaur1) [inline src/app/pages/fleet/assets/add-assets/add-assets.component.html]: replace the message with - Licence plate number must be between 7-8 alphanumeric characters.

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html [hunk 2]
```diff
@@ -57,16 +57,15 @@ <h4 class="text-4 mt-0 mb-0 text-dark">{{pageTitle}}</h4>
...
                             </div>
                             <div *ngIf="dName.invalid && (dName.dirty || dName.touched)" class="text-danger">
                               <div *ngIf="dName.errors.required">
-                                Employee ID is required.
+                                Employee id is required.
                               </div>
                               <div *ngIf="dName.errors.pattern">
-                                Employee ID must not contain special character. Employee ID must be
-                                between 2 to 30 characters and contain letters and numbers only.
+                                Employee id should be of minimum 4 characters which can be a combination of alphanumeric only.
```
Comment (FH-sandeepkaur1) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: message format = field_name can only contains alphanumeric characters with length between 4-120.

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html [hunk 3]
```diff
@@ -77,15 +76,14 @@ <h4 class="text-4 mt-0 mb-0 text-dark">{{pageTitle}}</h4>
                             <div class="input-group">
                               <input [(ngModel)]="driverData.employeeContractorId" name="employeeContractorId"
                                 type="text" class="form-control" placeholder="Any Numbers or Name" #cName="ngModel"
-                                required pattern="^[a-zA-Z0-9\s\-]{2,30}$" maxlength="30">
+                                required pattern="^[a-zA-Z0-9\s]{4,30}$" maxlength="30">
                             </div>
                             <div *ngIf="cName.invalid && (cName.dirty || cName.touched)" class="text-danger">
                               <div *ngIf="cName.errors.required">
-                                Contractor ID is required.
+                                Contractor id is required.
```
Comment (FH-sandeepkaur1) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: ID should be capital

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html [hunk 4]
```diff
@@ -142,11 +140,11 @@ <h4 class="text-4 mt-0 mb-0 text-dark">{{pageTitle}}</h4>
                           maxlength="30">
                         <div *ngIf="uName.invalid && (uName.dirty || uName.touched)" class="text-danger">
                           <div *ngIf="uName.errors.required">
-                            UserName is required.
+                            Username is required.
                           </div>
                           <div *ngIf="uName.errors.pattern">
-                            Username must contain small letters,numbers and special
-                            characters(period(.)) only.
+                            Username must contain small letters,numbers and special characters(period(.)) only.
```
Comment (FH-sandeepkaur1) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: message format = field_name must contain alphanumeric and allowed characters(. ,#)

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html [hunk 5]
```diff
@@ -224,7 +222,7 @@ <h4 class="text-4 mt-0 mb-0 text-dark">{{pageTitle}}</h4>
                                   Password is required.
                                 </div>
                                 <div *ngIf="pwd.errors.pattern">
-                                  Password must be 8 digits long and  a combination of small letters,capital letters and atleast one special characters(#,@,$,&).
+                                  Password must be a minimum of 8 digits long and a combination of small letters, capital letters,numbers and at least one special character (#,@,$,&).
                                 </div>
```
Comment (FH-sandeepkaur1) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.html:227]: message = Password must be 8 digits long and a combination of small letters,capital letters and atleast one special characters(#,@,$,&).

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html [hunk 6]
```diff
@@ -667,8 +665,7 @@ <h4 class="text-4 mt-0 mb-0 text-dark">{{pageTitle}}</h4>
                               Document# is required.
                             </div>
                             <div *ngIf="docNum.errors.pattern">
-                              Document# must contain combination of small letters,capital letters
-                              ,alphanumeric and atleast one special characters(#,/,-,_,.)
+                              Document# must contain a combination of letters, capital letters, numbers, and at least one special character (#,/,-,_,.)
```
Comment (FH-sandeepkaur1) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: change it please

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html [hunk 7]
```diff
@@ -679,11 +676,11 @@ <h4 class="text-4 mt-0 mb-0 text-dark">{{pageTitle}}</h4>
                           <input [(ngModel)]="document.issuingAuthority"
                             [attr.name]="'documentDetails['+i+'].issuingAuthority'"
                             [ngModelOptions]="{standalone: true}" type="text" class="form-control"
-                            placeholder="eg. Ontario,canada" #authority="ngModel" required pattern="^[a-zA-z\,]+$">
+                            placeholder="eg. Ontario,canada" #authority="ngModel" pattern="^[a-zA-z\s\,]+$">
                           <div *ngIf="authority.invalid && (authority.dirty || authority.touched)" class="text-danger">
 
                             <div *ngIf="authority.errors.pattern">
-                              Issuing Authority must contain letter and special characters(period(,)) only .
+                              Issuing authority must contain letter and special characters(period(,)) only .
```
Comment (FH-sandeepkaur1) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: message format = field_name must contain letters, numbers and allowed characters(. ,#)

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html [hunk 8]
```diff
@@ -811,7 +809,7 @@ <h4 class="text-4 mt-0 mb-0 text-dark">{{pageTitle}}</h4>
                           pattern="^[a-zA-Z0-9]{2,10}$" maxlength="10">
                         <div *ngIf="aci.invalid && (aci.dirty || aci.touched)" class="text-danger">
                           <div *ngIf="aci.errors.pattern">
-                            ACI ID must contain ony 10 numbers and combination of uppercase,lowercase and numbers only.
+                            ACI id must contain 10 alphanumeric characters only.
```
Comment (FH-sandeepkaur1) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: ID must be capital

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html [hunk 9]
```diff
@@ -853,8 +851,8 @@ <h4 class="text-4 mt-0 mb-0 text-dark">{{pageTitle}}</h4>
                             #fastCard="ngModel" pattern="^[4270|4110]{4}[0-9]{8}[00|01]{2}$" maxlength="14">
                           <div *ngIf="fastCard.invalid && (fastCard.dirty || fastCard.touched)" class="text-danger">
                             <div *ngIf="fastCard.errors.pattern">
-                              Fast Card must contain only 14 numbers beginning with 4270 or 4110 and ending with 00 or
-                              01.
+                              Fast Card must contain only 14 numbers beginning with 4270 or 4110 and ending with 00 or  01.
```
Comment (FH-sandeepkaur1) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: replace or with slash

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html [hunk 10]
```diff
@@ -1034,7 +1032,7 @@ <h4 class="text-4 mt-0 mb-0 text-dark">{{pageTitle}}</h4>
                         </ng-select>
                         <div *ngIf="paySlt.invalid && (paySlt.dirty || paySlt.touched)" class="text-danger">
                           <div *ngIf="paySlt.errors.required">
-                            Default Payment Type is required.
+                            Default payment Type is required.
```
Comment (FH-sandeepkaur1) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: Please check uppercase and lower case

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html [hunk 11]
```diff
@@ -1467,10 +1576,18 @@ <h4 class="text-4 mt-0 mb-0 text-dark">{{pageTitle}}</h4>
...
-                    <button type="submit" class="cus-btn-padd btn btn-success" *ngIf="!driverID"
-                     >Save</button>
-                    <button type="button" (click)="updateDriver()" *ngIf="driverID" class="btn cus-btn-padd btn-success"
-                      >Update</button>
+                                             
+                    <!-- if there is some error in validation  -->
+                    <button type="submit" class="cus-btn-padd btn btn-success" *ngIf="!driverID && !driverF.form.valid" [disabled]="!driverF.form.valid" >Save</button>
+                    
+                    <!-- if validations are ok and now the button will disable on submit click  --> 
+                    <button type="submit" class="cus-btn-padd btn btn-success" *ngIf="!driverID && driverF.form.valid" [disabled]="submitDisabled" >Save</button>
```
Comment (FH-sandeepkaur1) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.html:1590]: Please check add driver function on click

**Review by FH-sandeepkaur1 — COMMENTED** [review]
please make the required changes

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html (hunk 9 above)
Comment (nameetjambla) [reply src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: done

File: src/app/pages/fleet/assets/add-assets/add-assets.component.html (hunk 1 above)
Comment (nameetjambla) [reply src/app/pages/fleet/assets/add-assets/add-assets.component.html]: done

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html (hunk 2 above)
Comment (nameetjambla) [reply src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: done

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html (hunk 3 above)
Comment (nameetjambla) [reply src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: done

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html (hunk 4 above)
Comment (nameetjambla) [reply src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: done

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html (hunk 6 above)
Comment (nameetjambla) [reply src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: done

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html (hunk 8 above)
Comment (nameetjambla) [reply src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: done

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.html (hunk 10 above)
Comment (nameetjambla) [reply src/app/pages/fleet/drivers/add-driver/add-driver.component.html]: done

**Comment by nameetjambla** [comment]
> please make the required changes

done with the given as well as overall changes

**Review by FH-sandeepkaur1 — APPROVED** [review]

**Review by kunalfleethawks — APPROVED** [review]
Approved by @FH-sandeepkaur1 and @Hardeep8587

**Pull request merged**